Dockerfile

```diff
@@ -0,0 +1,42 @@
+# Use the official Node.js image as a base for building the application
+FROM node:22.12.0 AS build
+
+# Set the working directory inside the container
+WORKDIR /app
+
+# Copy package.json and package-lock.json
+COPY package*.json ./
+
+# Install all dependencies
+RUN npm install
+
+# Copy the entire application code including prisma folder
+COPY prisma ./prisma/
+COPY . .
+
+# Generate Prisma client
+RUN npm run prisma:generate
+
+# Build the NestJS application
+RUN npm run build
+
+# Create a smaller production image using the slim version of Node.js
+FROM node:22.12.0
```

Comment:
**suggestion**: Consider using a slim version of the Node.js image for the production stage to reduce the image size.